src/button/button.scss

```diff
@@ -28,57 +29,79 @@ $iui-button-padding-large: $iui-xs * 6;
   white-space: nowrap;
   border: 1px solid transparent;
   @media (prefers-reduced-motion: no-preference) {
-    transition: color $iui-speed-fast ease-out,
-      background-color $iui-speed-fast ease-out,
+    transition: color $iui-speed-fast ease-out, background-color $iui-speed-fast ease-out,
       border-color $iui-speed-fast ease-out;
   }
 
-  > .iui-icon:not(.iui-user-icon) {
-    width: $iui-icons-default;
-    height: $iui-icons-default;
-    @media (prefers-reduced-motion: no-preference) {
-      transition: fill $iui-speed-fast ease-out;
+  @include themed {
+    color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-2));
+  }
+
+  > .iui-button-icon {
+    @include themed {
+      fill: t(iui-icons-color-actionable);
     }
   }
 
-  &:enabled:hover {
+  &:hover,
+  &:focus-visible {
     text-decoration: none;
+    color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-1));
+
+    > .iui-button-icon {
+      @include themed {
+        fill: t(iui-icons-color-actionable-hover);
+      }
+    }
   }
 
   @include iui-focus;
+
+  &.iui-small {
+    @include iui-button-sizes(small);
+  }
+
+  &.iui-large {
+    @include iui-button-sizes(large);
+  }
+
   @include iui-button-sizes;
+
+  &[disabled] {
```

Comment:
This selector needs to be fixed (add `:disabled` and maybe even `[aria-disabled]='true'` in all places)

Reply:
Added `:disabled`